{
  "category": "Microservices",
  "name": "Microservices_20250523_173707",
  "description": "Infrastructure failure scenario for Microservices",
  "duration": 1596,
  "metadata": {
    "generated_at": "2025-05-23T17:37:07.797210",
    "phrase_templates_used": 15,
    "total_events": 7,
    "severity_distribution": {
      "low": 0,
      "medium": 0,
      "high": 3
    }
  },
  "events": [
    {
      "timestamp": "2025-05-23T17:37:07.797072",
      "event_type": "trigger",
      "description": "Initial failure detected: a service",
      "injection_spec": null
    },
    {
      "timestamp": "2025-05-23T17:38:08.797072",
      "event_type": "timeout",
      "description": "Timeout: docker network",
      "injection_spec": {
        "type": "timeout",
        "target": "auth-service",
        "severity": "high",
        "duration": 171,
        "log_snippet": "[2025-05-23 17:37:07] ERROR auth-service: docker network - operation failed",
        "metrics": {
          "error_rate": 0.6328761238743847,
          "response_time_ms": 8742,
          "cpu_usage": 0.7842295793632614,
          "memory_usage": 0.9099957765187872
        }
      }
    },
    {
      "timestamp": "2025-05-23T17:38:31.797072",
      "event_type": "failure",
      "description": "Failure: to service",
      "injection_spec": {
        "type": "failure",
        "target": "notification-service",
        "severity": "high",
        "duration": 130,
        "log_snippet": "[2025-05-23 17:37:07] ERROR notification-service: to service - operation failed",
        "metrics": {
          "error_rate": 0.7329354339024852,
          "response_time_ms": 9019,
          "cpu_usage": 0.9996313438651789,
          "memory_usage": 0.9460855567924737
        }
      }
    },
    {
      "timestamp": "2025-05-23T17:41:11.797072",
      "event_type": "timeout",
      "description": "Timeout: docker network",
      "injection_spec": {
        "type": "timeout",
        "target": "user-service",
        "severity": "high",
        "duration": 205,
        "log_snippet": "[2025-05-23 17:37:07] ERROR user-service: docker network - operation failed",
        "metrics": {
          "error_rate": 0.2825652850653424,
          "response_time_ms": 4255,
          "cpu_usage": 0.8745206590246765,
          "memory_usage": 0.9694895249930195
        }
      }
    },
    {
      "timestamp": "2025-05-23T17:41:41.797072",
      "event_type": "recovery",
      "description": "Recovery: the queue",
      "injection_spec": null
    },
    {
      "timestamp": "2025-05-23T17:42:03.797072",
      "event_type": "recovery",
      "description": "Recovery: the container",
      "injection_spec": null
    },
    {
      "timestamp": "2025-05-23T18:03:43.797072",
      "event_type": "recovery",
      "description": "System recovered: system restored",
      "injection_spec": null
    }
  ]
}